src/DMLQueryBuilder.php

```diff
@@ -34,35 +31,34 @@ final class DMLQueryBuilder extends AbstractDMLQueryBuilder
      * @throws InvalidConfigException
      * @throws NotSupportedException
      */
-    public function batchInsert(string $table, array $columns, iterable|Generator $rows, array &$params = []): string
+    public function batchInsert(string $table, array $columns, iterable $rows, array &$params = []): string
     {
         if (empty($rows)) {
             return '';
         }
 
-        if (($tableSchema = $this->schema->getTableSchema($table)) !== null) {
-            $columnSchemas = $tableSchema->getColumns();
-        } else {
-            $columnSchemas = [];
-        }
-
-        $mappedNames = $this->getNormalizeColumnNames($table, $columns);
         $values = [];
+        $columns = $this->getNormalizeColumnNames('', $columns);
+        $columnSchemas = $this->schema->getTableSchema($table)?->getColumns() ?? [];
 
-        /** @psalm-var array<array-key, array<array-key, string>> $rows */
+        /** @psalm-var array[] $rows */
         foreach ($rows as $row) {
+            $i = 0;
             $placeholders = [];
-            foreach ($row as $index => $value) {
-                if (isset($columns[$index], $mappedNames[$columns[$index]], $columnSchemas[$mappedNames[$columns[$index]]])) {
-                    /** @var mixed $value */
-                    $value = $this->getTypecastValue($value, $columnSchemas[$mappedNames[$columns[$index]]]);
+            /** @psalm-var mixed $value */
```

Comment:
Add this to psalm configuration:

```
<issueHandlers>
    <MixedAssignment errorLevel="suppress" />
</issueHandlers>
```

It's allow to do not add annotation for mixed values. [Example](https://github.com/yiisoft/package-template/blob/master/psalm.xml).